zkevm-circuits/src/evm_circuit/execution/balance.rs

```diff
@@ -0,0 +1,302 @@
+        let is_warm = cb.query_bool();
+        cb.account_access_list_write(
+            tx_id.expr(),
+            from_bytes::expr(&address.cells),
+            1.expr(),
+            is_warm.expr(),
+            Some(&mut reversion_info),
+        );
+
+        let exists = cb.query_bool();
+        let balance = cb.condition(exists.expr(), |cb| {
+            let balance = cb.query_cell();
+            cb.account_read(
+                from_bytes::expr(&address.cells),
+                AccountFieldTag::Balance,
+                balance.expr(),
+            );
+            balance
+        });
+        cb.condition(1.expr() - exists.expr(), |cb| {
+            cb.account_read(
+                from_bytes::expr(&address.cells),
+                AccountFieldTag::NonExisting,
+                0.expr(),
+            );
+        });
+
+        cb.stack_push(select::expr(exists.expr(), balance.expr(), 0.expr()));
+
+        let gas_cost = is_warm.expr() * GasCost::WARM_ACCESS.expr()
+            + (1.expr() - is_warm.expr()) * GasCost::COLD_ACCOUNT_ACCESS.expr();
+
+        let step_state_transition = StepStateTransition {
+            rw_counter: Delta(cb.rw_counter_offset()),
+            program_counter: Delta(1.expr()),
+            stack_pointer: Delta(0.expr()),
+            gas_left: Delta(-gas_cost),
+            reversible_write_counter: Delta(1.expr()),
+            ..Default::default()
+        };
+
+        let opcode = cb.query_cell();
+        let same_context = SameContextGadget::construct(cb, opcode, step_state_transition);
+
+        Self {
+            same_context,
+            address,
+            reversion_info,
+            tx_id,
+            is_warm,
+            balance,
+            exists,
+        }
+    }
+
+    fn assign_exec_step(
+        &self,
+        region: &mut CachedRegion<'_, '_, F>,
+        offset: usize,
+        block: &Block<F>,
+        tx: &Transaction,
+        call: &Call,
+        step: &ExecStep,
+    ) -> Result<(), Error> {
+        self.same_context.assign_exec_step(region, offset, step)?;
+
+        let address = block.rws[step.rw_indices[0]].stack_value().to_address();
+        let mut address_bytes = address.0;
+        address_bytes.reverse();
+        self.address.assign(region, offset, Some(address_bytes))?;
+
+        self.tx_id
+            .assign(region, offset, Value::known(F::from(tx.id as u64)))?;
+
+        self.reversion_info.assign(
+            region,
+            offset,
+            call.rw_counter_end_of_reversion,
+            call.is_persistent,
+        )?;
+
+        let (_, is_warm) = block.rws[step.rw_indices[4]].tx_access_list_value_pair();
+        self.is_warm
+            .assign(region, offset, Value::known(F::from(is_warm)))?;
+
+        let exists = step.account_exists(&address);
+        let balance = if exists {
+            block.rws[step.rw_indices[5]]
+                .table_assignment_aux(block.randomness)
+                .value
```

Comment:
I'm unsure why do we need `block.randomness` to set the balance. Is it passed directly as an RLC??

Reply:
I checked function [table_assignment_aux](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/zkevm-circuits/src/witness/rw.rs#L334). But it seems that only `value` is needed here, so I replaced it with `value_assignment(block.randomness)`.

And reference function [value_assignment](https://github.com/privacy-scaling-explorations/zkevm-circuits/blob/main/zkevm-circuits/src/witness/rw.rs#L518), it seems that the argument `randomness` is used in function `random_linear_combine` as below:
```
AccountFieldTag::CodeHash | AccountFieldTag::Balance => {
    RandomLinearCombination::random_linear_combine(value.to_le_bytes(), randomness)
}
```

Reply:
Hi @CPerezz , could you help review this again when you have time? Thanks.

Reply:
Ye sorry @silathdiir I missed this!!!